### ISSUE TITLE
Merge breaks when merging a node with a sequence.

### ISSUE THREAD

**Issue by ArcticSnowman** [perlpunk/YAML-PP-p5#36]
Given a node that is a sequence
```
node1: &node1
  - item1
  - item2
  - item3

node2:
  <<: [ *node1 ]
```

I get the following error: 
```
Expected hash for merge key at centos7/v5.18.2/lib/perl5/YAML/PP/Constructor.pm line 160, <STDIN> line 1.
 at centos7/v5.18.2/lib/perl5/YAML/PP/Loader.pm line 92.
```
---
node1: &node1
  item1: {}
  item2: {}
  item3: {}

node2:
  <<: [ *node1 ]
```
Works for node that are hashes.

**Comment by perlpunk** [comment]
That's expected behaviour. Merge keys only work for hashes.

**Comment by ArcticSnowman** [comment]
https://yaml.org/type/merge.html
```
 If the value associated with the merge key is a sequence, then this sequence is expected to contain mapping nodes and each of these nodes is merged in turn according to its order in the sequence.
```
So an array of scalars can't be merge?    That would be nice if it was possible.

**Comment by perlpunk** [comment]
```
<<: [ *node1 ]
```
here the "value associated with the merge key" is already a sequence. But the actual nodes in that sequence have to be mappings.
Merging sequences is not possible with this feature.

There are ideas how to make that possible in the next YAML version.
But also I might add a custom tag to YAML::PP that allows merging sequences.

**Comment by ArcticSnowman** [comment]
Ah,   that sequence..

**Comment by ArcticSnowman** [comment]
See https://metacpan.org/pod/YAML::LoadBundle  for ideas..

**Comment by perlpunk** [comment]
I was rather thinking about using tags. The merge key for mappings is also basically a tag, it's just implicit.
Explicit it would be: `!!merge <<: ...`

Something like:
```
seq1: &s1 [a, b, c]
seq2: &s2 [d, e, f]
merged: !merge-seq [*s1, *s2]
```
Since YAML has tags, there's no need to use special named scalars and structures.

**Comment by ArcticSnowman** [comment]
Yeah I believe our LoadBundle pre-dates the `merge` feature in YAML.   Might suggest to the authors of the LoadBundle to consider adding support for a  "TAG" based approach as well as a custom key based approach.
